packages/replay/src/replay.ts

```diff
@@ -534,17 +534,16 @@ export class ReplayContainer implements ReplayContainerInterface {
       // when an error occurs. Clear any state that happens before this current
       // checkout. This needs to happen before `addEvent()` which updates state
       // dependent on this reset.
-      if (this.recordingMode === 'error' && event.type === 2) {
+      if (this.recordingMode === 'error' && isCheckout) {
```

Comment:
Will this impact the behavior of start/stop recording in any way? I believe we should have test coverage around it

Reply:
So right now it will, because we do not always have `isCheckout === true` on initial snapshots. So we would need to to merge https://github.com/getsentry/rrweb/pull/56 in first to make this work as expected, I think.